### PR TITLE
Add endianness to serializers and deserializers

### PR DESCRIPTION
**Pull request by danlrobertson**
Enforce endianness in the creation of our serializers and deserializers
in the ipc module.

Resolves: #154

### CODE REVIEW

**Comment by antrik** [comment]
I'm not sure I understand. Is this working around a regression introduced by a change in the compiler? If so, is it really a good idea to work around it, rather than waiting for a fix in the compiler?...

File: src/ipc.rs [hunk 1]
```diff
@@ -152,7 +153,7 @@ impl<T> IpcSender<T> where T: Serialize {
                 let os_ipc_shared_memory_regions;
                 let os_ipc_channels;
                 {
-                    let mut serializer = bincode::Serializer::new(&mut bytes);
+                    let mut serializer = bincode::Serializer::<_, LittleEndian>::new(&mut bytes);
```
Comment (antrik) [inline src/ipc.rs]: Shouldn't that rather be `NativeEndian`?

(Which is apparently hard-aliased to `LittleEndian` anyway, for whatever reason -- but still it seems a more obvious choice conceptually?...)

**Comment by jdm** [comment]
This is the bincode crate publishing https://github.com/TyOverby/bincode/commit/5cdf2a248bc482da97e3bb091d9d459c1be983ac as a non-semver-compatible version update, presumably (https://github.com/TyOverby/bincode/commit/eebd6a98c3e8fc5c2b8bac2feaf9abf2b1683b1f).

**Comment by antrik** [comment]
Actually, if using `NativeEndian` -- and if that was indeed what the name suggests -- this would in fact be a good change regardless, since there is totally no reason to enforce any particular endianness here, and using the native one could be a tick more efficient on some machines. Otherwise... I'm not sure. Forcing little endian is probably still good, since it's the better choice in *most* cases, and it might not be default.

**Comment by antrik** [comment]
@jdm well, I don't quite understand the interaction between `serde` and `bincode`; but if it's indeed a `bincode` issue (and not compiler-related, as https://github.com/servo/ipc-channel/issues/154 suggests), I would assume it's caused by https://github.com/TyOverby/bincode/pull/103 -- but that was *supposed* to be API-neutral AIUI, in which case the regression should be fixed in `bincode` I guess?...

**Comment by antrik** [comment]
(For the record, even if unintentional, strictly speaking this *is* actually semver-compatible, since semver explicitly states that there are no API guarantees whatsoever for pre-releases... But then again, the Cargo interpretation of semver doesn't match the actual standard anyway.)

File: src/ipc.rs (hunk 1 above)
Comment (danlrobertson) [inline src/ipc.rs]: Ah, right... `NativeEndian` is much better.

**Comment by danlrobertson** [comment]
>  I don't quite understand the interaction between serde and bincode; but if it's indeed a bincode issue (and not compiler-related, as #154 suggests)

It is indeed a `bincode`/`serde` issue. I did not describe the issue well. I also do not know `bincode`/`serde` well.

**Comment by antrik** [comment]
Actually, this workaround will work *only* with the breaking API change; so the patch at least has to update the `bincode` version requirement in `Cargo.toml`... However, when the regression is (hopefully) fixed upstream, it won't work with later versions either. (And the broken version presumably gets yanked, so in fact the workaround won't work with *any* version once that happens...)

In other words, this can probably be only addressed in a sane way in `bincode` itself -- which is really the right thing to do anyway. In the free software world, we generally can, and should, fix problems at their source, rather then working around them...

**Comment by antrik** [comment]
See https://github.com/TyOverby/bincode/pull/128

**Pull request closed without merging**